src/container_runner.py

```diff
@@ -18,6 +19,37 @@
 DOCKER_DAEMON_CONFIG_PATH = Path("/etc/docker/daemon.json")
 
 
+def _obtain_tls(email: str, domain: str):
+    # Install Certbot for managing certificates
+    if email == "" or domain == "":
+        logger.warning("Skipping TLS setup, both an email and domain are required.")
+        return
+    try:
+        cache = snap.SnapCache()
+        certbot = cache["certbot"]
+        if not certbot.present:
+            certbot.ensure(snap.SnapState.Latest)
+    except snap.SnapError as e:
+        logger.error("An exception occurred when installing charmcraft. Reason: %s", e.message)
```

Comment:
this should say "while installing certbot" I think?

Reply:
Ah, yes! Thanks :smile: